raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2882,6 +2908,7 @@ private FetchRequestData buildFetchRequest() {
                 .setLastFetchedEpoch(log.lastFetchedEpoch())
                 .setFetchOffset(log.endOffset().offset())
                 .setReplicaDirectoryId(quorum.localDirectoryId())
+                .setHighWatermark(quorum.highWatermark().map(LogOffsetMetadata::offset).orElse(-1L))
```

Comment:
We don't need to condition this on the MV since we're relying on the resolved ApiVersions. Right?

Reply:
Correct. The version is negotiated using ApiVersions response and the field is ignorable. Since the field is ignorable, this fetch partition object can be serialized to version 18 and version 17.